### PR TITLE
Make deprecated functions noisier/defunct

### LINKED ISSUE #6387: Lifecycle updates
## Pre 1.0.0

* [x] `funs()` to defunct. Since 0.8.0 (Feb 2019).
* [x] `select_vars()` and friends to defunct. Since 0.8.4 (Oct 2019)

## Probably low-risk 1.0.0

* [x] Move `bench_tbls()` to defunct. GitHub code search suggests no CRAN users.
* [x] `location()` to defunct. Can't easily search for, but likely to be few uses.

## Increase warning frequency

* [x] `progress_estimated()` -> `deprecate_warn()`. Make internal version that `do()` can continue to use.
* [x]  Everything else 1.0.0 and older we should switch to warning unconditionally?

### PR DESCRIPTION
**Pull request by hadley**
Fixes #6387

### CODE REVIEW

File: R/defunct.r [hunk 1]
```diff
@@ -1,21 +1,119 @@
 #' Defunct functions
 #'
-#' `r lifecycle::badge("deprecated")`
+#' @description
+#' `r lifecycle::badge("defunct")`
 #'
-#' Executing these functions will tell you which function replaces them.
+#' These function were all deprecated for at least two years and are now
+#' defunct. Executing them to tell you which function to use instead.
```
Comment (DavisVaughan) [inline R/defunct.r:7]:
```suggestion
#' defunct. Executing them will tell you which function to use instead.
```

File: R/defunct.r [hunk 2]
```diff
@@ -1,21 +1,119 @@
...
+#' @export
+#' @rdname defunct
+eval_tbls2 <- function(tbls_x, tbls_y, op) {
+  lifecycle::deprecate_stop("1.0.0", "eval_tbls2()")
+}
+
+#' @export
+#' @rdname defunct
+location <- function(df) {
+  lifecycle::deprecate_stop("1.0.0", "location()", "lobst::ref()")
```
Comment (DavisVaughan) [inline R/defunct.r:112]:
```suggestion
  lifecycle::deprecate_stop("1.0.0", "location()", "lobstr::ref()")
```

File: tests/testthat/_snaps/defunct.md [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+      current_vars()
+    Condition
+      Error:
+      ! `current_vars()` was deprecated in dplyr 0.8.4 and is now defunct.
+      Please use `tidyselect::peek_vars()` instead.
+    Code
+      bench_tbl()
+    Condition
+      Error in `bench_tbl()`:
+      ! could not find function "bench_tbl"
```
Comment (DavisVaughan) [inline tests/testthat/_snaps/defunct.md:58]: READ YOUR SNAPSHOTS HADLEY

**Review by DavisVaughan — APPROVED** [review]
Woo!

File: tests/testthat/_snaps/defunct.md (hunk 3 above)
Comment (hadley) [reply tests/testthat/_snaps/defunct.md:58]: NOT MARKING EACH ERROR IS TOTALLY SAFE

**Pull request merged**